### PR TITLE
Improve Variable Font support

### PR DESCRIPTION
**Pull request by josh-hadley**
Updates to several freetype-py modules to improve support for Variable Fonts. Includes tests for added code, as well as an example demonstrating variable font usage.

### CODE REVIEW

File: freetype/__init__.py [hunk 1]
```diff
@@ -18,13 +18,22 @@
 import io
 import sys
 from ctypes import *
```
Comment (rougier) [inline freetype/__init__.py]: No connected to the PR but at some point we should get rid of this `from ctypes import *` in favor of `import ctypes`

**Comment by rougier** [comment]
Really nice ! Thanks for the PR. This looks good to me. Let's wait a day or two for potential comments and then we merge (maybe you'll need to remind me).

**Comment by HinTak** [comment]
FWIW, "from ctype import *" and "import ctypes" do different things. In the latter, you still need to do "ctypes.x" in code, vs bare "x" in the former.

File: freetype/__init__.py (hunk 1 above)
Comment (josh-hadley) [reply freetype/__init__.py]: Agree. There are lots of other code style/PEP8 & related issues that would be good to address (and maybe enforce, via CI) eventually. I didn't want to pollute this PR with such updates though 😄

**Comment by josh-hadley** [comment]
> FWIW, "from ctype import *" and "import ctypes" do different things. In the latter, you still need to do "ctypes.x" in code, vs bare "x" in the former.

Right. My preference would be `from ctypes import (<sequence of each individual thing used from ctypes>)`, so that only the import statement(s) would change, rather than each line of code where a ctypes item is used. We can discuss more in a later PR 😄

**Comment by rougier** [comment]
@HinTak Yes but I (personally) prefer to prefix, maybe because I'm using numpy a lot and, from time to time, I also need the math module with a lot of overlap/conflicts. Anyway, I can live with the `import *` if everyone's ok.

**Comment by HinTak** [comment]
I like what @josh-hadley says, an explicit "from ctypes import x, y, z" ... as long as the list does not get too long. But no need to change what's currently working...

Not a big fan of numpy - I think if you name something the same as a more common package's, you better be 100% compatible...

**Comment by madig** [comment]
For what it's worth, there is https://pypi.org/project/autoflake/ and maybe https://github.com/quentinsf/dewildcard that can automatically expand star imports.

**Comment by madig** [comment]
Woops! You might want to run `pylint --disable=all --enable=E Lib` to find more typos and stuff.

**Comment by josh-hadley** [comment]
> Woops! You might want to run `pylint --disable=all --enable=E Lib` to find more typos and stuff.

Yeah, I have been using flake8, but had disabled a bunch of warnings to keep the noise down from issues in the existing code...this would've been caught by flake8. It would be good to have pylint or flake8 or similar running in CI to catch these on PR submit :smile:

Related: the appveyor CI failure does not appear to be caused by the most recent commit (https://github.com/rougier/freetype-py/commit/83d0c53255). Seems like a timeout attempting to retrieve build components.

**Comment by madig** [comment]
Hrm, yeah, we download FreeType and HarfBuzz release tarballs so we can build a static Python wheel. Not sure what's going on there. Needs to be updated anyway, in a separate PR...

Agree on a clean-up PR with linting and stuff.

**Comment by rougier** [comment]
Merged ! Thanks again for the nice PR.

**Pull request merged**